src/components/Layout/ChipBase.js

```diff
@@ -1,48 +1,94 @@
+import { DIMENSION_ID_ORGUNIT } from '@dhis2/analytics'
 import i18n from '@dhis2/d2-i18n'
 import cx from 'classnames'
 import PropTypes from 'prop-types'
 import React from 'react'
+import {
+    VALUE_TYPE_BOOLEAN,
+    VALUE_TYPE_TRUE_ONLY,
+    getConditions,
+} from '../../modules/conditions.js'
+import { DIMENSION_TYPE_PERIOD } from '../../modules/dimensionConstants.js'
 import { DimensionIcon } from '../MainSidebar/DimensionItem/DimensionIcon.js'
 import styles from './styles/Chip.module.css'
 
+const VALUE_TYPE_TRUE_ONLY_NUM_OPTIONS = 2
+const VALUE_TYPE_BOOLEAN_NUM_OPTIONS = 3
+
 // Presentational component used by dnd - do not add redux or dnd functionality
 
-const renderChipLabelSuffix = (items, numberOfConditions) => {
-    let itemsLabel = ''
-    if (items.length) {
-        itemsLabel = i18n.t('{{count}} selected', {
-            count: items.length,
+const getChipSuffix = ({
+    dimension,
+    conditions = {},
+    items = [],
+    metadata,
+}) => {
+    const { id, dimensionType, optionSet, valueType } = dimension
+
+    if (
+        (id === DIMENSION_ID_ORGUNIT ||
+            dimensionType === DIMENSION_TYPE_PERIOD) &&
+        !items.length
+    ) {
+        return ''
+    }
+
+    const all = i18n.t('all')
+
+    if (!conditions.condition && !conditions.legendSet && !items.length) {
+        return `: ${all}`
```

Comment:
For consistency, might be best to either stick to the `suffix =` path or the direct return path (e.g. `return : '${all}'` everywhere).
So you could either move the `let suffix...` up higher and use that here.. or remove the `let suffix...` part below and just do a direct return instead.

Reply:
went with "return"